docs/help.md

```diff
@@ -0,0 +1,241 @@
+
+You probably have called `cache.readFragment` with a GraphQL
+document that doesn't contain a main fragment.
+
+This error occurs when no main fragment can be found, because the `DocumentNode`
+is maybe empty or does not contain fragments.
+
+When you're calling a fragment method, please ensure that you're only passing one
+or more fragments in your GraphQL document.
+
+## (7) Can't generate a key for readFragment(...) <a id="7"></a>
+
+> Can't generate a key for readFragment(...).
+> You have to pass an `id` or `_id` field or create a custom `keys` config for `???`.
+
+You probably have called `cache.readFragment` with data that the cache can't generate a
+key for.
+
+This may either happen because you're missing the `id` or `_id` field or some other
+fields for your custom `keys` config.
+
+Please make sure that you include enough properties on your data so that `readFragment`
+can generate a key.
+
+## (8) Invalid resolver data <a id="8"></a>
+
+> Invalid resolver value: The resolver at `???` returned an invalid typename that
+> could not be reconciled with the cache.
+
+This error may occur when you provide a cache resolver for a field using `resolvers` config.
+
+The value that you returns needs to contain a `__typename` field and this field must
+match the `__typename` field that exists in the cache, if any. This is because it's not
+possible to return a different type for a single field.
+
+## (9) Invalid resolver value <a id="9"></a>
+
+> Invalid resolver value: The field at `???` is a scalar (number, boolean, etc),
+> but the GraphQL query expects a selection set for this field.
+
+The GraphQL query that has been walked contains a selection set at the place where
+your resolver is located.
+
+This means that a full entity object needs to be returned, but instead the cache
+received a number, boolean, or another scalar from your resolver.
+
+Please check that your resolvers return scalars where there's no selection set,
+and entities where there is one.
+
+## (10) writeOptimistic(...) was called with sth that isn't a mutation <a id="10"></a>
+
+> writeOptimistic(...) was called with an operation that is not a mutation.
+> This case is unsupported and should never occur.
+
+This should never happen, please open an issue if it does. This occurs when `writeOptimistic`
+attempts to write an optimistic result for a query or subscription, instead of a mutation.
+
+## (11) writeFragment(...) was called with an empty fragment <a id="11"></a>
+
+> writeFragment(...) was called with an empty fragment.
+> You have to call it with at least one fragment in your GraphQL document.
+
+You probably have called `cache.writeFragment` with a GraphQL
+document that doesn't contain a main fragment.
+
+This error occurs when no main fragment can be found, because the `DocumentNode`
+is maybe empty or does not contain fragments.
+
+When you're calling a fragment method, please ensure that you're only passing one
```

Comment:
one or more? I think we only accept the first in a single writeFragment method, no?

Reply:
added a clarification, since we do accept multiple fragments, but use the first to start reading/writing